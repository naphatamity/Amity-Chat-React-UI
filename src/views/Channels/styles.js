import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  height: inherit;
`;

export const ChannelsWrapper = styled.div`
  padding: 16px;
`;

export const ChannelsList = styled.ul`
  list-style: none;
  padding: 0;
`;

export const ChannelRow = styled.li`
  width: 100%;
  display: flex;
  flexdirection: row;
  height: 72px;
`;

export const ChannelTitle = styled(Link)`
  font-style: normal;
  font-weight: 600;
  font-size: 17px;
  line-height: 22px;
  margin: 0;
  letter-spacing: -0.41px;
  text-align: left;
  text-decoration: none;
  color: #292b32;
  display: inline-block;
  text-overflow: ellipsis;
  width: 100%;
  overflow: hidden;
`;

export const ChannelDetails = styled.div`
  border-bottom: 1px solid #ebecef;
  flex-grow: 1;
  display: flex;
  margin-bottom: 12px;
`;

export const ChannelDescription = styled.div`
  margin-left: 12px;
  margin-right: 12px;
  flex-grow: 1;
  max-width: 60%;
`;

export const RecentMessage = styled.div`
  font-style: normal;
  font-weight: normal;
  font-size: 15px;
  line-height: 20px;
  letter-spacing: -0.24px;
  margin: 2px 0 8px;
  color: #898e9e;
`;
export const ChannelMeta = styled.div`
  margin-left: auto;
  text-align: right;
`;

export const LastActivity = styled.div`
  margin-bottom: 10px;
  font-size: 13px;
  color: #898e9e;
`;

export const UnreadCounter = styled.div`
  background: #fa4d30;
  color: white;
  border-radius: 20px;
  height: 20px;
  padding: 0px 6.5px;
  // width: auto;
  font-size: 13px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  line-height: 13px;
  margin-bottom: 8px;
`;

export const ChannelMenu = styled.div`
  margin-left: auto;
`;

export const TitleBar = styled.div`
  display: flex;
  flex-flow: row;
  flex-grow: 1;
  align-items: center;
`;

export const Title = styled.h2`
  margin: 0;
  margin-left: 8px;
  font-style: normal;
  font-weight: 600;
  font-size: 17px;
  line-height: 22px;
  letter-spacing: -0.41px;
`;
