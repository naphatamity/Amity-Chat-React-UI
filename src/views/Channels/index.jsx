import { useEffect, memo } from 'react';

import { useConfig } from '../../components/Config';
import ChatHome from '../../components/ChatHome';

import { Wrapper } from './styles';

const Channels = memo(() => {
  const { client, connectClientAlt, customer } = useConfig();

  useEffect(() => {
    if (!client.currentUser) {
      // eslint-disable-next-line no-console
      console.log('connect user channels', customer);
      connectClientAlt({ userId: customer.email, displayName: customer.name });
    }
  }, [client]);

  return (
    <Wrapper>
      <ChatHome />
    </Wrapper>
  );
});
export default Channels;
