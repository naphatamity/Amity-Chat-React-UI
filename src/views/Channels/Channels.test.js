import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Channels from '.';

describe('<Channels /> spec', () => {
  it('renders the component', () => {
    const container = render(
      <Router>
        <Channels />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
