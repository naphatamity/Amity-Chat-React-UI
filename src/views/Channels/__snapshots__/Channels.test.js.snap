// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Channels /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa cQfXqj"
      >
        <div
          class="sc-egiSv bvyLrL"
        >
          <div
            class="sc-iqsfdx iToDHF"
          >
            <h2
              class="sc-crHlIS jiFitH"
            >
              Chat
            </h2>
          </div>
          <div
            class="sc-kDThTU jihaaO"
          >
            <a
              href="/channel/create"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment-alt-plus fa-w-16 "
                data-icon="comment-alt-plus"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288zM336 184h-56v-56c0-8.8-7.2-16-16-16h-16c-8.8 0-16 7.2-16 16v56h-56c-8.8 0-16 7.2-16 16v16c0 8.8 7.2 16 16 16h56v56c0 8.8 7.2 16 16 16h16c8.8 0 16-7.2 16-16v-56h56c8.8 0 16-7.2 16-16v-16c0-8.8-7.2-16-16-16z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="sc-gsDJrp eZKoEd"
        >
          <ul
            class="sc-dkPtyc eLDXiK"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa cQfXqj"
    >
      <div
        class="sc-egiSv bvyLrL"
      >
        <div
          class="sc-iqsfdx iToDHF"
        >
          <h2
            class="sc-crHlIS jiFitH"
          >
            Chat
          </h2>
        </div>
        <div
          class="sc-kDThTU jihaaO"
        >
          <a
            href="/channel/create"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-comment-alt-plus fa-w-16 "
              data-icon="comment-alt-plus"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288zM336 184h-56v-56c0-8.8-7.2-16-16-16h-16c-8.8 0-16 7.2-16 16v56h-56c-8.8 0-16 7.2-16 16v16c0 8.8 7.2 16 16 16h56v56c0 8.8 7.2 16 16 16h16c8.8 0 16-7.2 16-16v-56h56c8.8 0 16-7.2 16-16v-16c0-8.8-7.2-16-16-16z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="sc-gsDJrp eZKoEd"
      >
        <ul
          class="sc-dkPtyc eLDXiK"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
